### DISCUSSION TITLE
Navigation from Custom 404 Page to Home Not Working on Deployed Site

### DISCUSSION THREAD

**Discussion by lucifax** [vercel/community#5968] [Help]
### Description
I've encountered an issue with my Next.js application deployed on Vercel where navigation from my custom 404 page back to the home page (`/`) does not work as expected. Specifically, when I click the "Go Home" button or the logo on the 404 page, which both link to `/`, the URL in the browser address bar updates to indicate the home page, but the displayed content remains on the 404 page. This issue does not occur in my local development environment (localhost:3000), where navigation works perfectly.

### Steps to Reproduce
1. https://property-next-chi.vercel.app/
2. Navigate to a non-existent route to trigger the custom 404 page (e.g., `/non-existent-route`).
3. Click the "Go Home" button or the logo, both intended to redirect users to the home page (`/`).
4. Observe that while the URL updates to `/`, the content remains on the 404 page.

### Expected Behavior
Clicking the "Go Home" button or the logo from the 404 page should navigate the user back to the home page, displaying the home page's content.

### Actual Behavior
The URL updates to reflect the home page route (`/`), but the content of the 404 page remains displayed.

### Additional Context
- The issue only occurs on the deployed version of the site on Vercel; local development works as expected.
- My application uses standard Next.js `<Link>` components for navigation.
- Here's a simplified version of my 404 page component:

```jsx import React from "react";
import Link from "next/link";
import { FaExclamationTriangle } from "react-icons/fa";

const NotFoundPage = () => {
  return (
    <section className="bg-blue-50 min-h-screen flex-grow">
      <div className="container m-auto max-w-2xl py-24">
        <div className="bg-white px-6 py-24 mb-4 shadow-md rounded-md border m-4 md:m-0">
          <div className="flex justify-center">
            <FaExclamationTriangle className="fas fa-exclamation-triangle fa-5x text-8xl text-yellow-400" />
          </div>
          <div className="text-center">
            <h1 className="text-3xl font-bold mt-4 mb-2">Page Not Found</h1>
            <p className="text-gray-500 text-xl mb-10">
              The page you are looking for does not exist.
            </p>
            <Link
              href="/"
              className="bg-blue-700 hover:bg-blue-800 text-white font-bold py-4 px-6 rounded"
            >
              Go Home
            </Link>
          </div>
        </div>
      </div>
      <div className="flex-grow"></div>
    </section>
  );
};

export default NotFoundPage;


### Example

https://property-next-chi.vercel.app/

### Steps to Reproduce
1. https://property-next-chi.vercel.app/
2. Navigate to a non-existent route to trigger the custom 404 page (e.g., `/non-existent-route`).
3. Click the "Go Home" button or the logo, both intended to redirect users to the home page (`/`).
4. Observe that while the URL updates to `/`, the content remains on the 404 page.

**Comment by amyegan** [comment]
Hey @lucifax. I couldn't replicate this problem on a basic test app, but I may have made some incorrect assumptions about how your project is set up. If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), someone here might be able to recognize what is going wrong.
